radix-engine/src/blueprints/epoch_manager/validator.rs

```diff
@@ -47,6 +50,16 @@ pub struct ValidatorSubstate {
     /// Whether this validator is currently interested in participating in the consensus.
     pub is_registered: bool,
 
+    /// A fraction of the effective emission amount which gets transferred to the validator's owner
+    /// (by staking it and depositing the stake units to the [`locked_owner_stake_unit_vault_id`]).
+    /// Note: it is a decimal factor, not a percentage (i.e. `0.015` means "1.5%" here).
+    /// Note: it may be overridden by [`validator_fee_change_request`], if it contains a change
+    /// which already became effective.
+    pub validator_fee_factor: Decimal,
```

Comment:
I've now added the doc describing this.

(We only promote the value from the request when we are forced to, i.e. when another request comes. Technically we _could_ promote it as a drive-by [e.g. _also_ during `apply_emissions()`] - but it would be more code [possibly more complex], and no performance gain)

Reply:
If the whole substate is updated during `apply_emission`. I don't think there will be major overhead.

Basically, I was thinking about the dashboard and wallet use cases.  They are going to display the validator fee - they will need to read both `validator_fee_factor`, `validator_fee_change_request` and the `current_epoch` and the compute, in your model.

Reply:
Let me include @dhedey in case he has any thoughts on this.

Reply:
hmm, I thought that such dashboard would read the `validator_fee_change_request` anyways (since it is an interesting information about near future too). The `current_epoch` is also a very common read (I assume).

TBH, if I was implementing such dashboard, I'd actually prefer to pull the `validator_fee_change_request` and do the computation, just to make sure I display the up-to-date information.

Imagine a scenario:
- Validator X is in validator set
- The owner changes its fee to 10%, and the change becomes effective after the wait
- Stakers gradually unstake [because the fee is "uhhh rather high"]
- Validator X is dropped from the validator set because of too low stake
- so the owner changes its fee back to 0.5%, hoping that stakers come back. The change becomes effective on the next epoch.
- Stakers look at Validator X dashboard... and they still see 10% there!
  - This is because `apply_emission()` was not called for Validator X, because it is not in validator set and does not receive emissions.

Reply:
This all seems reasonable to me - we should just make sure we explain this to the dashboard team! 👍